dsgrid/config/project_config.py

```diff
@@ -289,6 +302,79 @@ def dimension_mapping_manager(self):
     def dimension_mapping_manager(self, val: DimensionMappingRegistryManager):
         self._dimension_mapping_mgr = val
 
+    def get_base_dimension(self, dimension: DimensionType):
+        """Return the base dimension matching dimension_type.
+
+        Parameters
+        ----------
+        dimension_type : DimensionType
+
+        Returns
+        -------
+        DimensionConfig
+
+        """
+        for key, dim_config in self.base_dimensions.items():
+            if key.type == dimension:
+                return dim_config
+        assert False, dimension
+
+    def get_supplemental_dimensions(self, dimension: DimensionType):
+        """Return the supplemental dimensions matching dimension (if any).
+
+        Parameters
+        ----------
+        dimension_type : DimensionType
+
+        Returns
+        -------
+        DimensionConfig
+
+        """
+        return [v for k, v in self.supplemental_dimensions.items() if k.type == dimension]
+
+    def get_base_to_supplemental_dimension_mappings_by_types(self, dimension: DimensionType):
```

Comment:
Everywhere we have dimension: DimensionType, shouldn't the name be dimension_type?

Reply:
I'm not sure I agree with the statement globally, but I do agree here. If the line length limit forced this to be
```
def get_base_to_supplemental_dimension_mappings_by_types(
    self,
    dimension: DimensionType
):
```
then I would want to use the shorter name.